05/DB.test.js

```diff
@@ -1 +1,151 @@
-import DB from './DB';
\ No newline at end of file
+import DB from "./DB";
+
+describe("class DB", () => {
+  it(`Creates new instance of the class | WHEN: "new" operator is used.`, () => {
+    const db = new DB();
+
+    function createNewDB(db) {
+      if (db) {
+        return true;
+      }
+
+      return false;
+    }
+
+    expect(createNewDB(db)).toBe(true);
+  });
+
+  describe("insert(data)", () => {
+    // scenario
+    it("REJECTS: the promise | WHEN: data ID is not a number", async () => {
+      // given
+      const db = new DB();
+      // Czy nie brakuje tutaj when?
```

Comment:
Możesz zrobić w takiej formie, że najpierw wprowadzasz dane (then) i sprawdzasz czy to co zostało zwrócone jest ok. Po prostu pominąłeś ten krok dla wygody/szybkości.

`const result = await db.insert({ id: "not a number", a: 1, b: 2 }`

PS. Pamiętaj o `async`, gdy używasz `await`